osc/osc_test.go

```diff
@@ -314,14 +315,20 @@ func TestReadPaddedString(t *testing.T) {
 		buf []byte // buffer
 		n   int    // bytes needed
 		s   string // resulting string
+		e   error  // expected error
 	}{
-		{[]byte{'t', 'e', 's', 't', 's', 't', 'r', 'i', 'n', 'g', 0, 0}, 12, "teststring"},
-		{[]byte{'t', 'e', 's', 't', 0, 0, 0, 0}, 8, "test"},
+		{[]byte{'t', 'e', 's', 't', 'S', 't', 'r', 'i', 'n', 'g', 0, 0}, 12, "testString", nil},
+		{[]byte{'t', 'e', 's', 't', 'e', 'r', 's', 0}, 8, "testers", nil},
+		{[]byte{'t', 'e', 's', 't', 's', 0, 0, 0}, 8, "tests", nil},
+		{[]byte{'t', 'e', 's', 't', 0, 0, 0, 0}, 8, "test", nil},
+		{[]byte{}, 0, "", io.EOF},
```

Comment:
This currently returns EOF, I think something suggested there may be no string, so should TestReadPaddedString catch this?

Reply:
I don't see why it shouldn't, that's the expected behavior, so technically, a deviation from that is bad.

Reply:
So we SHOULD expect this to fail with an EOL?

Reply:
Yes, this is expected behavior.